docs/docs/concepts/dialogue-understanding.mdx

```diff
@@ -0,0 +1,161 @@
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: LLMCommandGenerator
+    llm:
+      model_name: "gpt-4"
+      request_timeout: 7
+      temperature: 0.0
+# - ...
+```
+
+The `model_name` defaults to `gpt-4`. The model name should be set 
+to a chat model of [OpenAI](https://platform.openai.com/docs/guides/gpt/chat-completions-api).
+
+Similarly, you can specify the `request_timeout` and 
+`temperature` parameters for the LLM. The `request_timeout` 
+defaults to `7` seconds and the `temperature` defaults to `0.0`.
+
+If you want to use Azure OpenAI Service, you can configure the necessary
+parameters as described in the
+[Azure OpenAI Service](../llms/llm-configuration.mdx#additional-configuration-for-azure-openai-service)
+section.
+
+:::info Using Other LLMs
+
+By default, OpenAI is used as the underlying LLM provider. 
+
+The used LLM provider provider can be configured in the
+`config.yml` file to use another provider, e.g. `cohere`: 
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: LLMCommandGenerator
+    llm:
+      type: "cohere"
+# - ...
+```
+
+For more information, see the
+[LLM setup page on llms and embeddings](../llms/llm-configuration.mdx#other-llmsembeddings)
+
+:::
+
+### Customizing The Prompt
+
+Because the `LLMCommandGenerator` uses in-context learning, one of the primary ways
+to tweak or improve performance is to customize the prompt. 
+
+In most cases, you should be able to achieve what you need to by customizing the `description` fields
+in your flows.
+Every flow has its own `description` field, and optionally every `step` in your flow can have one as well.
+If you notice that a flow is being triggered when it shouldn't, or that a slot is not being extracted correctly,
+adding more detail to the description will often solve the issue. 
+
+For example if you have a `transfer_money` flow with a `collect` step for the slot `amount`, you might add a description
+to extract the value more reliably: 
+
+
+```yaml-rasa title="flows.yml"
+flows:
+  transfer_money:
+    description: |
+      This flow lets users send money to friends
+      and family, in US Dollars.
+    steps:
+      - id: "ask_recipient"
+        collect_information: recipient
+        next: "ask_amount"
+      - id: "ask_amount"
+        collect_information: amount
+        // highlight-next-line
+        description: the amount of money to send. extract only the numerical value, ignoring the currency.
+        next: "transfer_successful"
+```
+
+### Customizing The Prompt Template
+
+If you are unable to get something to work via editing your yaml files, you can go one level deeper
+and customise the prompt template used to drive the `LLMCommandGenerator`.
+To do this, you can write your own prompt as a jinja2 template and provide it to the component as a file:
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: LLMCommandGenerator
+    // highlight-next-line
```

Comment:
there's an [alternate syntax](https://docusaurus.io/docs/markdown-features/code-blocks#highlighting-with-metadata-string) that doesn't break our scripts (some tests are failing)